src/main/java/games/strategy/debug/ClientLogger.java

```diff
@@ -87,6 +88,16 @@ public static void logError(final @Nullable String msg, final Throwable e) {
     showErrorMessage(msg);
   }
 
+  /**
+   * Overload of {@link #logError(String, Throwable)} specialized for {@code ExecutionException}. An
+   * {@code ExecutionException} contains no useful information; it's simply an adapter to tunnel exceptions thrown by
+   * tasks through the {@code Executor} API. We only log the cause to reduce the number of stack trace frames visible
+   * to the user.
+   */
+  public static void logError(final @Nullable String msg, final ExecutionException e) {
```

Comment:
Why allow the String parameter to be `@Nullable`? Technically it is okay, but I don't think we ever want to see a null error message. I'm concerned the annotation would invite someone to write:
`logError(null, executionExceptionInstance)`

I'd like to see the `@Nullable` annotation to be dropped (unless I'm missing something important, but again I would hate to see the code that makes use of the nullable).

Grant it, logging is confused, but it is intended that calling this message is throwing an error message in the users face :smiling_imp:  -  A pop-up with error message: `null` seems like insult to injury.

Reply:
When I added the `@Nullable` annotations to this class a few months ago, I believe I was simply trying to capture the current behavior.  That is, the code did not prevent a `null` message and worked just fine if one was specified.

With that said, I agree `null` should never be passed for the message.  The only issue I see with adding a new `checkNotNull()` precondition at this point is that there are so many calls to the `ClientLogger` methods, it would take some time to go through them all and verify they don't pass `null`.

I suppose we could just add the precondition checks, accept that we may introduce an NPE regression at runtime, and just fix them as they are reported.  Is that acceptable?

For now, since there are only two callers of this overload, I'll add the preconditions here only.  8c9fdd5f6